process/rag_main.py

```diff
@@ -0,0 +1,44 @@
+from langchain_community.llms import Ollama
+from langchain_community.document_loaders import WebBaseLoader
+from langchain_community.embeddings import OllamaEmbeddings
+from langchain_community.vectorstores import Chroma
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_text_splitters import RecursiveCharacterTextSplitter
+from langchain.chains.combine_documents import create_stuff_documents_chain
+from langchain.chains import create_retrieval_chain
+
+# Invoke chain with RAG context
+llm = Ollama(model="phi3")
+## Load page content
+loader = WebBaseLoader("https://docs.smith.langchain.com/user_guide")
+docs = loader.load()
+print(docs)
+
+## Vector store things
+embeddings = OllamaEmbeddings(model="phi3")
+text_splitter = RecursiveCharacterTextSplitter()
+split_documents = text_splitter.split_documents(docs)
+vector_store = Chroma.from_documents(split_documents, embeddings)
+
+## Prompt construction
+prompt = ChatPromptTemplate.from_template(
+    """
+            Answer the following question only based on the given context
+                                                    
+            <context>
+            {context}
+            </context>
+                                                    
+            Question: {input}
+"""
+)
+
+## Retrieve context from vector store
+docs_chain = create_stuff_documents_chain(llm, prompt)
+retriever = vector_store.as_retriever()
+retrieval_chain = create_retrieval_chain(retriever, docs_chain)
+
+## Winner winner chicken dinner
+response = retrieval_chain.invoke({"input": "how can langsmith help with testing?"})
+print(":::ROUND 2:::")
+print(response["answer"])
```

Comment:
Can you help me create a function that allows input from the terminal so we can chat until the user decides to stop?